nucypher/blockchain/eth/utils.py

```diff
@@ -31,22 +31,33 @@ def datetime_to_period(datetime: maya.MayaDT, seconds_per_period: int) -> int:
     return int(future_period)
 
 
-def datetime_at_period(period: int, seconds_per_period: int) -> maya.MayaDT:
-    """Returns the datetime object at a given period, future, or past."""
-
-    now = maya.now()
-    current_period = datetime_to_period(datetime=now, seconds_per_period=seconds_per_period)
-    delta_periods = period - current_period
+def period_to_epoch(period: int, seconds_per_period: int) -> int:
+    epoch = period * seconds_per_period
+    return epoch
+
+
+def datetime_at_period(period: int, seconds_per_period: int, start_of_period: bool = False) -> maya.MayaDT:
+    """
+    Returns the datetime object at a given period, future, or past.
+    If start_of_period, the datetime object represents the first second of said period.
+    """
+    if start_of_period:
+        datetime_at_start_of_period = maya.MayaDT(epoch=period_to_epoch(period, seconds_per_period))
+        return datetime_at_start_of_period
+    else:
+        now = maya.now()
+        current_period = datetime_to_period(datetime=now, seconds_per_period=seconds_per_period)
+        delta_periods = period - current_period
 
-    # +
-    if delta_periods:
-        target_period = now + maya.timedelta(days=delta_periods)
+        # +
+        if delta_periods:
+            target_datetime = now + maya.timedelta(days=delta_periods)
 
-    # -
-    else:
-        target_period = now - maya.timedelta(days=delta_periods)
+        # -
```

Comment:
Wait... `else` works only when `delta_periods == 0`!

Did you mean simply always `now + timedelta(days=delta_periods)`? Timedelta understands negative time too.

Also why days when we have `seconds_per_period` (what if it's not 1 day)? Maybe `maya.timedelta(seconds=1) * seconds_per_period * delta_periods` or something?

Reply:
+1 on the use of seconds instead

Reply:
opened https://github.com/nucypher/nucypher/issues/1435